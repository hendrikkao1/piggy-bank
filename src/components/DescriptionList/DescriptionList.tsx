interface DescriptionListProps {
  items: [string, React.ReactNode][];
}

export const DescriptionList = ({ items }: DescriptionListProps) => (
  <dl className="divide-y divide-zinc-900/20">
    {items.map(([term, description]) => (
      <div key={term} className="py-6 grid grid-cols-2 gap-2">
        <dt className="text-sm font-semibold text-zinc-900">{term}</dt>
        <dd className="text-sm text-zinc-900 col-span-1">{description}</dd>
      </div>
    ))}
  </dl>
);
