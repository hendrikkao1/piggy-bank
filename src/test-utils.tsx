import React from "react";
import {
  RenderHookOptions,
  RenderOptions,
  render,
  renderHook,
} from "@testing-library/react";
import { NextIntlClientProvider } from "next-intl";
import messages from "./messages/en-US.json";

export const AllTheProviders = ({
  children,
}: {
  children: React.ReactNode;
}) => (
  <NextIntlClientProvider locale={"en-US"} messages={messages}>
    {children}
  </NextIntlClientProvider>
);

const customRender = (ui: React.ReactElement, options: RenderOptions = {}) =>
  render(ui, { wrapper: AllTheProviders, ...options });

const customRenderHook = <TProps, TResult>(
  callback: (props: TProps) => TResult,
  options?: RenderHookOptions<TProps>,
) => renderHook(callback, { wrapper: AllTheProviders, ...options });

export * from "@testing-library/react";

export { customRender as render, customRenderHook as renderHook };
