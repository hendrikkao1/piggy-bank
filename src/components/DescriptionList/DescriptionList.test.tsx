import { render } from "@/test-utils";
import { DescriptionList } from "./DescriptionList";

describe("DescriptionList", () => {
  it("renders", () => {
    const { container } = render(
      <DescriptionList items={[["term", "description"]]} />,
    );

    expect(container).toMatchSnapshot();
  });

  it("renders react nodes", () => {
    const { container } = render(
      <DescriptionList
        items={[["term", <span key="description">description</span>]]}
      />,
    );

    expect(container).toMatchSnapshot();
  });
});
