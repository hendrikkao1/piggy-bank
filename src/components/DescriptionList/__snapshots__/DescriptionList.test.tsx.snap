// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DescriptionList renders 1`] = `
<div>
  <dl
    class="divide-y divide-zinc-900/20"
  >
    <div
      class="py-6 grid grid-cols-2 gap-2"
    >
      <dt
        class="text-sm font-semibold text-zinc-900"
      >
        term
      </dt>
      <dd
        class="text-sm text-zinc-900 col-span-1"
      >
        description
      </dd>
    </div>
  </dl>
</div>
`;

exports[`DescriptionList renders react nodes 1`] = `
<div>
  <dl
    class="divide-y divide-zinc-900/20"
  >
    <div
      class="py-6 grid grid-cols-2 gap-2"
    >
      <dt
        class="text-sm font-semibold text-zinc-900"
      >
        term
      </dt>
      <dd
        class="text-sm text-zinc-900 col-span-1"
      >
        <span>
          description
        </span>
      </dd>
    </div>
  </dl>
</div>
`;
